Autoincrements northwind.sql

ixtza opened the pull request:
- on table's id: employees, us_states, suppliers, shippers, region, products, orders, categories
- add sequence for id
- alter each table's id by `SET DEFAULT` to nextval("table id's sequence")

ixtza committed:
New changes at northwind.sql:
- on table's id: employees, us_states, suppliers, shippers, region, products, orders, categories
- add sequence for id
- alter each table's id by `SET DEFAULT` to nextval("table id's sequence")

jundi77 commented:
Jadi ada masalah kalau kita mau idnya hingga lebih dari satu juta, karena datatype di tabel masih smallint. Tolong diupdate menggunakan datatype `SERIAL`, mendapat keuntungan autoincrements yang dimanage secara internal. Mohon sequencenya disesuaikan, karena sepertinya tidak digunakan setelah mengganti datatype menjadi `SERIAL`.

jundi77 reviewed (changes requested):
Mohon tinjau komen sebelumnya

ixtza committed:
Autoincrements northwind.sql:
-on table's id: employees, us_states, suppliers, shippers, region, products, orders, categories
-each table pk's id type is now SERIAL
-foreign key with SMALLINT DATA TYPE is now INTEGER
-with SEQUENCE + ALTER TABLE, altering the next value of serial accordingly

jundi77 commented on database/northwind.sql:
```diff
@@ -3912,7 +3912,6 @@ ALTER TABLE ONLY employees
     START 10
-    INCREMENT 1
     OWNED BY employees.employee_id;
 
```
Sequence dihapus saja, kan tidak dipakai?

ixtza replied:
Ok

ixtza committed:
Autoincrements northwind.sql:
- remove all sequence
- modify ALTER COLUMN id's

jundi77 commented on database/northwind.sql:
```diff
@@ -3906,144 +3906,69 @@ ALTER TABLE ONLY employees
 
-ALTER TABLE ONLY public.employees
-	ALTER COLUMN employee_id SET DEFAULT nextval('employee_id_seq');
+ALTER TABLE ONLY employees
+	ALTER COLUMN employee_id SET DEFAULT nextval(pg_get_serial_sequence('employees','employee_id')) + 1;
```
`nextval` jangan diatur lagi, sudah internal oleh pgsql :)

tidak diatur lagi untuk menghindari konflik config yang mungkin tidak terlihat, sehingga sulit didebug :)

ixtza replied:
ok

ixtza committed:
Autoincrements northwind.sql
- romove all ALTER COLUMN for table's id

jundi77 reviewed (approved):
lgtm :+1:
